jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/PathResourceFactory.java

```diff
@@ -13,19 +13,63 @@
 
 package org.eclipse.jetty.util.resource;
 
+import java.io.IOException;
 import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.FileSystemNotFoundException;
+import java.nio.file.FileSystems;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.nio.file.Paths;
+import java.util.Collections;
 
 public class PathResourceFactory implements ResourceFactory
 {
+    static final boolean ENABLE_GRAALVM_RESOURCE_SCHEME = (System.getProperty(
+        "org.graalvm.nativeimage.kind") != null);
```

Comment:
Instead of a System Property, is there a class/resource that can be requested that will return a URL with scheme `resource:` that will work on all GraalVM runtimes?

That way startup could look for it, and auto-register a unique `GraalVMResourceFactory`, if the returned URL is scheme `resource`.

Reply:
The system property is set for all GraalVM Native-Image environments (i.e., binary executable or shared library). Since it's a property, it may also be set for testing purposes in a non-native (HotSpot) environment. The code is written in a way that should not break anything in that case (notwithstanding the potential OSGI issue with the resource:/ scheme that Oracle probably has to figure out somehow)

Reply:
I've added this check upon request because @joakime mentioned that OSGI would use `resource:` for other purposes. I don't know exactly how always registering `resource:` would be a problem though. It's of equivalent standard as `jrt`, which also isn't always available.

Moreover, checking for the availability of classes in Native-Image comes at the cost of adding support for reflection, which needs additional configuration. I'd be happy to remove that check. @gregw mentioned earlier that there could be an interest to support arbitrary schemes / add discovery of new ones.